db/db_impl/db_impl_files.cc

```diff
@@ -312,9 +312,26 @@ void DBImpl::FindObsoleteFiles(JobContext* job_context, bool force,
         // logs_ could have changed while we were waiting.
         continue;
       }
-      // This WAL file is not live, so it's OK if we never sync the rest of it
-      // or if we close it *after* removing from `logs_`. If it's already
-      // closed, then it's been fully synced.
+      // This WAL file is not live, so it's OK if we never sync the rest of it.
+      // If it's already closed, then it's been fully synced. If
+      // !background_close_inactive_wals then we need to Close it before
```

Comment:
Maybe explain why we need to close the WAL file before removing from `logs_` here. It's not very clear to me why we need to do this. My guess is it's because checkpoint may link WALs that are not in `logs_` and this ensure any WAL not in `logs_` is closed.

Reply:
That explanation is on the comment for background_close_inactive_wals. If wanting to know why "If !background_close_inactive_wals then we need to" that seems like a natural place to look.